src/Transport.ts

```diff
@@ -947,22 +954,30 @@ export class Transport extends EnhancedEventEmitter<TransportEvents>
 				// Clear pending pause Consumer map.
 				this._pendingPauseConsumers.clear();
 
+				const localIds = pendingPauseConsumers
+					.map((consumer) => consumer.localId);
+
+				if (localIds.length === 0)
+				{
+					logger.debug('_pausePendingConsumers() | there is no consumer to be paused');
```

Comment:
"Consumer"

Reply:
This can be moved to the beginning for the method to avoid superfluous `map()` exection, for clarity:

 ```ts
if (this._pendingPauseConsumers.size === 0)
{
  logger.debug('_pausePendingConsumers() | there is no consumer to be paused');

  return;
}
```

Same with the rest of places where this is done.